.scripts/dependencies.ts

```diff
@@ -0,0 +1,361 @@
+        const entryPath: string = resolvePath(folderPath, entryName);
+        const entryStats: fs.Stats = fs.lstatSync(entryPath);
+        if (entryStats.isDirectory()) {
+          deleteFolder(entryPath);
+        } else {
+          deleteFile(entryPath);
+        }
+      }
+      fs.rmdirSync(folderPath);
+    } else {
+      throw error;
+    }
+  }
+}
+
+/**
+ * Read the contents of text file at the provided filePath.
+ * @param {string} filePath The path to the text file to read.
+ * @returns {string} The text contents of the text file at the provided filePath.
+ */
+function readTextFileContents(filePath: string): string {
+  return fs.readFileSync(filePath, { encoding: "utf8" });
+}
+
+/**
+ * Execute the provided command on the shell synchronously.
+ * @param {string} command The command to execute.
+ * @param {string} workingDirectory The working directory to execute the command in.
+ * @returns {void}
+ */
+function execute(command: string, workingDirectory: string): void {
+  log(workingDirectory, `Running "${command}"...`);
+  execSync(command, { cwd: workingDirectory, stdio: [0, 1, 2] });
+}
+
+/**
+ * Get the absolute path to this repository's folder path.
+ * @returns {string} The absolute path to this repository's folder path.
+ */
+export function getThisRepositoryFolderPath(): string {
+  return resolvePath(__dirname, "..");
+}
+
+/**
+ * Get the absolute path to the package.json in this repository.
+ * @returns {string} The absolute path to the package.json.
+ */
+function getPackageJsonFilePath(packageFolder?: string): string {
+  return resolvePath(packageFolder, "package.json");
+}
+
+/**
+ * Get the absolute path to the local clone of the repository with the provided name.
+ * @param {string} repoName The name of the repository.
+ * @returns {string} The absolute path to the local clone of the repository.
+ */
+export function getLocalRepositoryPath(repoName: string): string {
+  return resolvePath(getThisRepositoryFolderPath(), "..", repoName);
+}
+
+/**
+ * Get the package.json file contents parsed as a JSON object.
+ * @param {string=} packageJsonFilePath The path to the package.json file to read. If this is not
+ * provided, then the package.json file at the root of this repository will be used.
+ * @returns {{}} The parsed package.json file contents.
+ */
+function getPackageJson(packageJsonFilePath: string): any {
+  return JSON.parse(readTextFileContents(packageJsonFilePath));
+}
+
+/**
+ * Get the dependencies from the provided dependencies dictionary that have local clones.
+ * @param {{ [packageName: string]: string }} dependencies A dictionary of package names to package
+ * versions.
+ * @param {string[]} clonedRepositoryNames The array to put the names of the local cloned
+ * repositories into.
+ * @returns {void}
+ */
+function getClonedRepositories(dependencies: { [packageName: string]: string }): string[] {
+  const clonedRepositoryNames: string[] = [];
+  if (clonedRepositoryNames && dependencies) {
```

Comment:
if dependencies can be falsy maybe it would be good to mark the parameter as nullable (note that `{}` isn't falsy)